app/src/main/java/com/simply/birthdayapp/commonpresentation/components/textfield/InputTextField.kt

```diff
@@ -0,0 +1,125 @@
+package com.simply.birthdayapp.commonpresentation.components.textfield
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.input.OffsetMapping
+import androidx.compose.ui.text.input.TransformedText
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.simply.birthdayapp.R
+import com.simply.birthdayapp.commonpresentation.theme.TextFieldContainerColor
+import com.simply.birthdayapp.commonpresentation.theme.TextFieldErrorContainerColor
+import com.simply.birthdayapp.commonpresentation.theme.TextFieldPlaceholderStyle
+import com.simply.birthdayapp.commonpresentation.theme.TextFieldShape
+import com.simply.birthdayapp.commonpresentation.theme.TextFieldTextStyle
+
+class PasswordAsteriskVisualTransformation : VisualTransformation {
+    override fun filter(text: AnnotatedString): TransformedText {
+        val asterisks = AnnotatedString("*".repeat(text.length))
+        return TransformedText(asterisks, OffsetMapping.Identity)
+    }
+}
+
+@Composable
+fun InputTextField(
+    modifier: Modifier = Modifier,
+    value: MutableState<String>,
+    error: MutableState<String>,
+    placeholder: String = "",
+    isHasError: Boolean = false,
+    isPassword: Boolean = false,
+    leadingIcon: @Composable (() -> Unit)? = null,
+    trailingIcon: @Composable (() -> Unit)? = null,
+) {
+    var passwordVisible by remember { mutableStateOf(false) }
+
+    Box(modifier = modifier) {
+        BasicTextField(
+            value = value.value,
+            onValueChange = { value.value = it },
+            singleLine = true,
+            textStyle = TextFieldTextStyle.copy(textAlign = TextAlign.Start),
+            modifier = Modifier
+                .fillMaxWidth()
+                .then(modifier),
```

Comment:
looks suspicious, why not write modifier.fillMaxWidth ?

Reply:
I didn't notice, maybe it did auto-completed 
removed